crates/oxc_transformer/src/utils.rs

```diff
@@ -50,3 +51,120 @@ pub trait CreateVars<'a> {
         }
     }
 }
+
+pub const RESERVED_WORDS_ES3_ONLY: [&str; 24] = [
+    "abstract",
+    "boolean",
+    "byte",
+    "char",
+    "double",
+    "enum",
+    "final",
+    "float",
+    "goto",
+    "implements",
+    "int",
+    "interface",
+    "long",
+    "native",
+    "package",
+    "private",
+    "protected",
+    "public",
+    "short",
+    "static",
+    "synchronized",
+    "throws",
+    "transient",
+    "volatile",
+];
+
+const RESERVED_WORD_STRICT: [&str; 9] = [
+    "implements",
+    "interface",
+    "let",
+    "package",
+    "private",
+    "protected",
+    "public",
+    "static",
+    "yield",
+];
+
+pub const KEYWORDS: [&str; 35] = [
+    "break",
+    "case",
+    "catch",
+    "continue",
+    "debugger",
+    "default",
+    "do",
+    "else",
+    "finally",
+    "for",
+    "function",
+    "if",
+    "return",
+    "switch",
+    "throw",
+    "try",
+    "var",
+    "const",
+    "while",
+    "with",
+    "new",
+    "this",
+    "super",
+    "class",
+    "extends",
+    "export",
+    "import",
+    "null",
+    "true",
+    "false",
+    "in",
+    "instanceof",
+    "typeof",
+    "void",
+    "delete",
+];
+
+/// https://github.com/babel/babel/blob/ff3481746a830e0e94626de4c4cb075ea5f2f5dc/packages/babel-helper-validator-identifier/src/identifier.ts#L85-L109
+pub fn is_identifier_name(name: &Atom) -> bool {
+    let string = name.as_str();
+    if string.is_empty() {
+        return false;
+    }
+    let mut is_first = true;
+    for ch in string.chars() {
+        if is_first {
+            is_first = false;
+            if !is_id_start(ch) {
+                return false;
+            }
+        } else if !is_id_continue(ch) {
+            return false;
+        }
+    }
+    true
+}
+
+pub fn is_valid_identifier(name: &Atom, reserved: bool) -> bool {
+    if reserved && (KEYWORDS.contains(&name.as_str()) || is_strict_reserved_word(name, true)) {
+        return false;
+    }
+    is_identifier_name(name)
+}
+
+pub fn is_strict_reserved_word(name: &Atom, in_module: bool) -> bool {
+    is_reserved_word(name, in_module) || RESERVED_WORD_STRICT.contains(&name.as_str())
+}
```

Comment:
Since it will check every identifier in the object key, is it will be better to use `Set` here?